package.json

```diff
@@ -27,15 +27,25 @@
     "grunt-svgmin": "^3.1.2",
     "grunt-usemin": "^3.1.1",
     "grunt-wiredep": "^2.0.0",
+    "jasmine": "^2.4.1",
+    "jasmine-core": "^2.4.1",
     "jshint-stylish": "^2.1.0",
+    "karma": "^1.1.1",
+    "karma-jasmine": "^1.0.2",
+    "karma-junit-reporter": "^1.1.0",
+    "karma-phantomjs-launcher": "^1.0.1",
+    "karma-spec-reporter": "0.0.26",
     "load-grunt-tasks": "^3.4.0",
     "time-grunt": "^1.3.0"
   },
   "engines": {
     "node": ">=0.10.0"
   },
   "scripts": {
-    "test": "grunt test"
+    "postinstall": "bower install",
+    "pretest": "npm install",
+    "test": "karma start test/karma.conf.js",
+    "test-single-run": "karma start test/karma.conf.js --single-run"
```

Comment:
@kwk There are no dumb questions. Use `npm run test-single-run` (note `run` in there). `npm test` is a special case, but in general you need to use `npm run` (or `npm run-script`) to execute the `scripts` that you define in `package.json`.

```
❯ npm run test-single-run

> docker-registry-frontend@0.0.2 test-single-run /Users/marca/dev/git-repos/docker-registry-frontend_2
> karma start test/karma.conf.js --single-run

19 07 2016 06:29:24.145:WARN [karma]: Port 8080 in use
19 07 2016 06:29:24.148:WARN [karma]: Port 8081 in use
19 07 2016 06:29:24.148:INFO [karma]: Karma v1.1.1 server started at http://localhost:8082/
19 07 2016 06:29:24.149:INFO [launcher]: Launching browser PhantomJS with unlimited concurrency
19 07 2016 06:29:24.157:INFO [launcher]: Starting browser PhantomJS
19 07 2016 06:29:24.613:INFO [PhantomJS 2.1.1 (Mac OS X 0.0.0)]: Connected on socket /#L0dobW7vCDANyy35AAAA with id 59001397

  docker-registry-frontend
    ✓ /home should display home page
    ✓ /repositories should display repository list page
    ✓ /repositories/10 should display repository list page
    ✓ /repositories/20 should display repository list page
    ✓ URL with repositoryUser and repositoryName and no tagsPerPage should display repository detail page
    ✓ URL with repositoryUser and repositoryName and tagsPerPage should display repository detail page
    ✓ URL with repositoryName but no repositoryUser and no tagsPerPage should display repository detail page
    ✓ URL with repositoryName but no repositoryUser and tagsPerPage should display repository detail page
    ✓ /about should display about page
    ✓ /tag/repositoryUser/repositoryName/latest should display tag detail page
    ✓ /image/88e37c7099fa should display image detail page
    ✓ /image/88e37c7099fa/tag should display create tag page
    ✓ /unknown-url should display home page

  RepositoryListController
    ✓ should attach some keys to the scope

  MainController
    ✓ should attach an appVersion and registryHost to the scope

PhantomJS 2.1.1 (Mac OS X 0.0.0): Executed 15 of 15 SUCCESS (0.008 secs / 0.077 secs)
TOTAL: 15 SUCCESS
```